### ISSUE TITLE
Multiple tomcat instance behind a load balancer

### ISSUE THREAD

**Issue by seenu5788** [redisson/redisson#2470]
Redisson version: 

redisson-tomcat-8-3.11.6.jar
redisson-all-3.11.6.jar

Hi, We are trying to host the application in 2 tomcat instance hosted behind a application gateway(load balancer in AWS). In both tomcat instance we have configured redisson session manager with required jars in tomcat/lib. 

Below is the context xml entry:

<Manager className="org.redisson.tomcat.RedissonSessionManager" configPath="${catalina.base}/redisson.yaml" readMode="MEMORY" updateMode="AFTER_REQUEST" broadcastSessionEvents="true"/>

Scenario:

1) Bring up instance 1 and login to app, go to a screen and do some changes and don't save yet. This ensures we have logged in to instance 1.

2) Bring up instance 2 now. Bring down the instance 1. Try to do save on the application now and app throws below error.

[redisson-netty-5-14] ERROR org.redisson.client.handler.CommandDecoder - Unable to decode data. channel: [id: 0x621c0852, L:/172.31.4.234:47394 - R:172.31.5.89/172.31.5.89:6379], reply: ReplayingDecoderByteBuf(ridx=138706, widx=138838), command: (HGETALL), params: [redisson:tomcat_session:24BFE948F528FB8C7035901297D535D2]
java.io.IOException: java.lang.NullPointerException
        at org.nustaq.serialization.FSTObjectInput.readObject(FSTObjectInput.java:247)
        at org.redisson.codec.FstCodec$1.decode(FstCodec.java:250)
        at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:378)
        at org.redisson.client.handler.CommandDecoder.decodeList(CommandDecoder.java:422)
        at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:387)
        at org.redisson.client.handler.CommandDecoder.decodeCommand(CommandDecoder.java:209)
        at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:147)
        at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:117)
        at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:503)
        at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:366)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:281)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1422)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:931)
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:700)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:635)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:552)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:514)
        at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1050)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException
        at org.nustaq.serialization.FSTObjectInput.readObjectWithHeader(FSTObjectInput.java:357)
        at org.nustaq.serialization.FSTObjectInput.readObjectFields(FSTObjectInput.java:713)
        at org.nustaq.serialization.FSTObjectInput.instantiateAndReadNoSer(FSTObjectInput.java:566)
        at org.nustaq.serialization.FSTObjectInput.readObjectWithHeader(FSTObjectInput.java:374)
        at org.nustaq.serialization.FSTObjectInput.readObjectInternal(FSTObjectInput.java:331)
        at org.nustaq.serialization.serializers.FSTMapSerializer.instantiate(FSTMapSerializer.java:78)
        at org.nustaq.serialization.FSTObjectInput.instantiateAndReadWithSer(FSTObjectInput.java:501)
        at org.nustaq.serialization.FSTObjectInput.readObjectWithHeader(FSTObjectInput.java:370)
        at org.nustaq.serialization.FSTObjectInput.readObjectFields(FSTObjectInput.java:713)
        at org.nustaq.serialization.FSTObjectInput.readObjectCompatibleRecursive(FSTObjectInput.java:635)
         at org.nustaq.serialization.FSTObjectInput.readObjectCompatible(FSTObjectInput.java:574)
        at org.nustaq.serialization.FSTObjectInput.instantiateAndReadNoSer(FSTObjectInput.java:559)
        at org.nustaq.serialization.FSTObjectInput.readObjectWithHeader(FSTObjectInput.java:374)
        at org.nustaq.serialization.FSTObjectInput.readObjectFields(FSTObjectInput.java:713)
        at org.nustaq.serialization.FSTObjectInput.instantiateAndReadNoSer(FSTObjectInput.java:566)
        at org.nustaq.serialization.FSTObjectInput.readObjectWithHeader(FSTObjectInput.java:374)
        at org.nustaq.serialization.FSTObjectInput.readObjectInternal(FSTObjectInput.java:331)
        at org.nustaq.serialization.FST

ObjectInput.readObject(FSTObjectInput.java:311)
        at org.nustaq.serialization.FSTObjectInput.readObject(FSTObjectInput.java:245)
        ... 26 more

Basically we are trying to aim for zero downtime with these 2 instances, so that if one instance goes down, other instance should be able to pick up the application state and continue the execution without user needing to re-login.

Please help us with the above error.

**Comment by mrniko** [comment]
try to use `org.redisson.codec.SerializationCodec`
